### PR TITLE
Fix paligemma detection inference

### LINKED ISSUE #31425: use_cache=False makes a huge difference in Paligemma
### System Info

Hello I think something is not right with Paligemma, I figured out the logloss was great but my accuracy or other metrics related to generated sequence were bad.
I realized generating tokens one by one improved performance, and then I realized .generate gave different results (even on the example given from the HF page of Paligemma).
This could be more easily observed with complex problems, where the differences are bigger. @molbap @ArthurZucker 

```from transformers import AutoProcessor, PaliGemmaForConditionalGeneration
from PIL import Image
import requests
import torch
 
model_id = "google/paligemma-3b-mix-224"
 
url = "https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/transformers/tasks/car.jpg?download=true"
image = Image.open(requests.get(url, stream=True).raw)
 
model = PaliGemmaForConditionalGeneration.from_pretrained(model_id).eval()
processor = AutoProcessor.from_pretrained(model_id)
 
# Instruct the model to create a caption in Spanish
prompt = "segment wall"
model_inputs = processor(text=prompt, images=image, return_tensors="pt")
input_len = model_inputs["input_ids"].shape[-1]
 
with torch.inference_mode():
    print('generate automatically')
    generation = model.generate(**model_inputs, max_new_tokens=100, do_sample=False)
    generation = generation[0][input_len:]
    decoded = processor.decode(generation, skip_special_tokens=True)
    print(decoded)
    
    print('generate manually token per token')
    while True:
        pred = model(input_ids= model_inputs["input_ids"], pixel_values=model_inputs['pixel_values'],
                      attention_mask=torch.ones_like(model_inputs["input_ids"]), ).logits
        new_tok = torch.argmax(pred[0,-1])
        model_inputs["input_ids"] = torch.cat([model_inputs["input_ids"], new_tok[None][None]], dim=1)
        decoded = processor.tokenizer.decode(model_inputs["input_ids"][0])
        print(decoded)
        if '<eos>' in decoded:
            break
```
PS: use_cache=False makes the problem disappear.

### Who can help?

_No response_

### Information

- [X] The official example scripts
- [X] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

1. Generate something difficult to generate (like segmentation of a hardly recognizable item like "segment wall" on the example in HF with the car)
2. Generate tokens one by one and see the difference

### Expected behavior

.generate by default should use greedy decoding I think, therefore in my opinion generating tokens one by one should give the same result, not better ones.

### PR DESCRIPTION
**Pull request by molbap**
# What does this PR do?

- Extends the extended attention mask of paligemma, fixing an inference bug especially for detection.
- Adds an object detection integration test to avoid future regressions

Fixes #31425 

hf hub discussion: https://huggingface.co/google/paligemma-3b-mix-448/discussions/6 

## Who can review?

@pcuenca @ArthurZucker

### CODE REVIEW

**Comment by molbap** [comment]
cc @zucchini-nlp, a generation bugfix you might want to look at

**Review by pcuenca — APPROVED** [review]
Verified it works, I can now replicate the same results with and without `use_cache`. Thank you! 🔥

**Review by ArthurZucker — APPROVED** [review]
thanks 🤗

**Comment by edmondja** [comment]
Is this fix already integrated to the last release https://github.com/huggingface/transformers/releases ? I am not sure release notes are supposed to be exhaustive @ArthurZucker (PS: merci de représenter le MVA)

**Comment by ArthurZucker** [comment]
Hey! 
image
this has been in the last 2 patch releases! They release notes for 4.42.0 should have it!

**Comment by ArthurZucker** [comment]
😉 sympa de croiser un alumni!

**Comment by ArthurZucker** [comment]
It is at the bottom: 
image

and absolutely no worries, we know this was quite important for everyone! 🤗
